test/semantics/select-case.f90

```diff
@@ -0,0 +1,172 @@
+! Test SELECT CASE Constraints: C1145, C1146, C1147, C1148, C1149
+program selectCaseProg
+implicit none
+   ! local variable declaration
+   character :: grade1 = 'B'
+   integer :: grade2 = 3
+   logical :: grade3 = .false.
+   character (len = 10) :: name = 'test'
+   real :: grade4 = 2.0
+   logical, parameter :: grade5 = .false.
+   CHARACTER(KIND=1), PARAMETER    :: ze = 'a', at='b'
+   CHARACTER(KIND=4), PARAMETER :: mh='c'
+
+   select case (grade1)
+      case ('A')
+      print*, "Excellent!"
+      case ('B')
+      case ('C')
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade2)
+      case (1)
+      print*, "Excellent!"
+      case (2)
+      case (3)
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case (.true.)
+       name = 'true'
+      case (.false.)
+        name = 'false'
+   end select
+   print*, "Your grade is ", name
+
+   select case (name)
+      case default
+         print*, "Invalid grade"
+      case ('now')
+      case ('test')
+         print*, "Well done"
+   end select
+
+   !ERROR: SELECT CASE expression must be of type CHARACTER, INTEGER, OR LOGICAL
+   select case (grade4)
+      case (1.0)
+      print*, "Excellent!"
+      case (2.0)
+      case (3.0)
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case default
+         print*, "Invalid grade"
+      case (.true.)
+         print*, "Well done"
+      !ERROR: Not more than one of the selectors of SELECT CASE statement may be default
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case default
+         print*, "Invalid grade"
+      case (.true.)
+      !ERROR: SELECT CASE value must be of type LOGICAL
+      case (3)
```

Comment:
This is referring to a case-stmt so it would be to say just "CASE value must be of type LOGICAL".

Reply:
Thanks Tim, I will do the necessary changes.